File: 0560/main.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        pres = {0:1};
        count, pre = 0, 0
        for n in nums:
            pre += n
            if pre - k in pres:
                count += pres[pre-k]
            if pre in pres:
                pres[pre] += 1
            else:
                pres[pre] = 1
        return count

File: 0560/main.rs
use std::collections::HashMap;

impl Solution {
    pub fn subarray_sum(nums: Vec<i32>, k: i32) -> i32 {
        let mut pres: HashMap<i32,i32> = HashMap::new();
        let mut pre: i32 = 0;
        let mut count: i32 = 0;
        pres.insert(0, 1);
        for n in nums.iter() {
            pre += n;
            if let Some(&m) = pres.get(&(pre-k)) {
                count += m;
            }
            pres.entry(pre).and_modify(|m| *m += 1).or_insert(1);
        }
        return count;
    }
}